src/test/lib/scheduler/scheduler_test.cpp

```diff
@@ -290,4 +318,23 @@ TEST_F(SchedulerTest, SingleWorkerGuaranteeProgress) {
   Hyrise::get().scheduler()->finish();
 }
 
+TEST_F(SchedulerTest, DetermineQueueIDForTask) {
+  if (std::thread::hardware_concurrency() < 2) {
+    GTEST_SKIP();
+  }
+
+  Hyrise::get().topology.use_fake_numa_topology(2, 1);
+  auto node_queue_scheduler = std::make_shared<NodeQueueScheduler>();
+  Hyrise::get().set_scheduler(node_queue_scheduler);
+
+  std::shared_ptr<AbstractTask> task = std::make_shared<JobTask>([&]() {});
```

Comment:
1. const?
2. up to you: if we want to more strictly use auto, you could also write
```c++
const auto task = std::static_pointer_cast<AbstractTask>(std::make_shared<JobTask>([&]() {}));
```